Headers/DebugServer2/Target/ThreadBase.h

```diff
@@ -57,6 +57,9 @@ class ThreadBase {
                            Address const &address = Address()) = 0;
 
 public:
+  virtual ErrorCode beforeResume();
```

Comment:
I'm trying to find a way to not have these be completely decorrelated
because every time you call resume, you have to remember to call
beforeResume. It adds complexity and is error-prone.

On Tuesday, August 9, 2016, Francis Ricci <notifications@github.com> wrote:

> In Headers/DebugServer2/Target/ThreadBase.h
> <https://github.com/facebook/ds2/pull/243#discussion_r74100032>:
>
> > @@ -57,6 +57,9 @@ class ThreadBase {
> >                             Address const &address = Address()) = 0;
> >
> >  public:
> > +  virtual ErrorCode beforeResume();
>
> The reason why it isn't in resume, though, is that the resume function is
> implemented in the platform-specific classes, not in the base class. And
> before-resume should always happen, regardless of platform. I suppose we
> could add a resume in the base class, which calls beforeResume and then
> the resume in the platform specific class, but I don't know that this is
> necessary/better
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/2a73b10db4691d10fd51d07ec46fbded7b36f6af..b32f1128fa8ceb0010cf5f79262d062aca477162#r74100032>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADHbDvFE63jYrPSO688GhtXXtWMtOYNks5qeLLTgaJpZM4H9FRy>
> .
>


-- 
Stephane Sezer